context/app/static/js/stores/useVisualizationStore.js

```diff
@@ -0,0 +1,26 @@
+import create from 'zustand';
+
+const useVisualizationStore = create((set) => ({
+  vizIsFullscreen: false,
+  vizEscSnackbarIsOpen: false,
+  setVizEscSnackbarIsOpen: (val) => set({ vizEscSnackbarIsOpen: val }),
+  expandViz: () => {
+    set({ vizIsFullscreen: true, vizEscSnackbarIsOpen: true });
+    document.onkeydown = function preventDefault(evt) {
+      if (evt.keyCode === 27) evt.preventDefault();
+    };
+  },
+  collapseViz: () => {
+    set({ vizIsFullscreen: false, vizEscSnackbarIsOpen: false });
+    document.onkeydown = null;
```

Comment:
Is it worth asking what happens if some other handler had been defined? or, better to keep it simple and cross that bridge when we come to it?